drivers/ieee802154/ieee802154_kw41z.c

```diff
@@ -24,6 +24,61 @@
 
 #include "fsl_xcvr.h"
 
+#if defined(CONFIG_NET_L2_OPENTHREAD)
+#include <net/openthread.h>
+#endif
+
+
+#if defined(CONFIG_NET_L2_OPENTHREAD)
+/*
+ * Frame Control Field and sequence number
+ * See Section 5.2.1.1
+ */
+/*
+ * The KW41Z 15.4 MAC needs to know if the packet being sent expects
+ * an 'ar' so it has to look at each packet to determine if the
+ * submission to the hardware needs to set the appropriate bit. The
+ * definition for this data structure is deep in the directory tree
+ * of the Zephyr 802.15.4 stack instead of the /include tree so when
+ * you enable OpenThread you no longer have access to the support
+ * routines that hid this detail.
+ *
+ * Ideally the 802.15.4 packet structure defs would just be in the
+ * normal /include tree and we could avoid all this stuff.
+ */
+struct ieee802154_fcf_seq {
+	struct {
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+		u16_t frame_type		:3;
+		u16_t security_enabled		:1;
+		u16_t frame_pending		:1;
+		u16_t ar			:1;
+		u16_t pan_id_comp		:1;
+		u16_t reserved			:1;
+		u16_t seq_num_suppr		:1;
+		u16_t ie_list			:1;
+		u16_t dst_addr_mode		:2;
+		u16_t frame_version		:2;
+		u16_t src_addr_mode		:2;
+#elif __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+		u16_t reserved			:1;
+		u16_t pan_id_comp		:1;
+		u16_t ar			:1;
+		u16_t frame_pending		:1;
+		u16_t security_enabled		:1;
+		u16_t frame_type		:3;
+		u16_t src_addr_mode		:2;
+		u16_t frame_version		:2;
+		u16_t dst_addr_mode		:2;
+		u16_t ie_list			:1;
+		u16_t seq_num_suppr		:1;
+#endif
+	} fc __packed;
+
+	u8_t sequence;
+} __packed;
+#endif
```

Comment:
Let me fix this the issue #5942 so you won't need this.

Reply:
I think the real fix for this is to just put the 802.15.4 packet framing definitions into the common directory tree. I know that in my previous PR you didn't want to do this because you felt this was internal information to the 15.4 native stack but I don't look at it that way. Even Linux has this information in a common include tree.

Reply:
I know Linux does it like that, but I really prefer to keep the crude details of framing into its current place. Also, if for some reason one wants to rewrite the framing logic or even replace the L2 by another one, it will be easier that way.

Reply:
We are talking about the framing details of an IEEE standard. That is why I would suggest having an ieee802154.h in the common include space that only has typedefs and stuff for the 15.4 protocol as defined in the standard.

Reply:
The way frames are parsed here is all but standard. You could do it in different ways.

Reply:
You control this part but your point about this being easier if we replace L2 isn't clear to me. All I'm talking about is the 802.15.4 protocol, which these drivers are providing HW access for. You are already adding this special part to the OT enablement to add in part of the code that is part of the native stack. The drivers should be agnostic to what is attaching to it and not need any additional special hooks to get to this information. My view is that the native 15.4 stack, OT, and any other protocol stack we may put on top should be as self contained as possible and we should avoid borrowing pieces of the stacks to accomplish what is really just an IEEE802.15.4 protocol detail and not tied to OT or our native 15.4 stack..

Reply:
I see, let me update #6631 I think I have a better compromise that will suit our both opinions: no frame headers being public at well as no specific part of the L2 being built for that.